docs/workflow-and-job.md

```diff
@@ -10,151 +10,222 @@ It can execute anything that can be executed from the Python code, for example:
     def __init__(self, id):
         self.id = id
-        self.retry_count = 20
-        self.retry_pause_sec = 100
 
     def run(self, runtime):
-        print(runtime)
+        print(f'Running job {self.id} at {runtime}')
+
+example_workflow = Workflow(
+    workflow_id='example_workflow',
+    definition=[Job('1'), Job('2')])
 
-job = SimpleJob('my_simple_job')
+example_workflow.run()
 ```
 
-## Workflow
+Output:
+```text
+Running job 1 at 2020-01-01
+Running job 2 at 2020-01-01
+```
 
-The `Workflow` class arranges jobs into a DAG. There are 2 ways of specifying job arrangement. First one is passing a list
-of jobs to `Workflow`:
+The second one is passing the `Definition` object, which allows you to create a graph.
 
+Let us say that we want to create the following DAG:
+    
+```
+       |--job2--|
+job1-->          -->job4 
+       |--job3--|
+```
+
+The implementation looks like this:
+
+[`./docs_examples/workflow_and_job/graph_workflow.py`](./docs_examples/workflow_and_job/graph_workflow.py)
 ```python
-from biggerquery.build import Workflow
+job1, job2, job3, job4 = Job('1'), Job('2'), Job('3'), Job('4')
 
-class Job(object):
-    def __init__(self, id):
-        self.id = id
-    
-    def run(self, runtime):
-        print(runtime)
+graph_workflow = Workflow(workflow_id='graph_workflow', definition=Definition({
+    job1: (job2, job3),
+    job2: (job4,),
+    job3: (job4,)
+}))
+graph_workflow.run()
+```
 
-example_workflow = Workflow(definition=[Job('1'), Job('2')])
+Output:
+```text
+Running job 1 at 2020-01-01
+Running job 2 at 2020-01-01
+Running job 3 at 2020-01-01
+Running job 4 at 2020-01-01
 ```
 
-The second one is passing `Definition` object, it allows you to create a graph:
+The `Workflow` class provides `run` and `run_job` methods. When you run a single job through a workflow object, 
+without providing the `runtime` parameter, the `Workflow` class will pass the current date-time (local time) as default.
 
+[`./docs_examples/workflow_and_job/run_workflow_and_job.py`](./docs_examples/workflow_and_job/run_workflow_and_job.py)
 ```python
-from biggerquery.build import Workflow
-from biggerquery.build import Definition
-from biggerquery.build import WorkflowJob
+simple_workflow = Workflow(
+    workflow_id='simple_workflow',
+    runtime_as_datetime=True,
+    definition=[Job('1')])
+simple_workflow.run_job('1')
+simple_workflow.run()
+simple_workflow.run_job('1', '1970-01-01')
+simple_workflow.run('1970-01-01')
+```
 
-class Job(object):
-    def __init__(self, id):
-        self.id = id
-    
-    def run(self, runtime):
-        print(runtime)
+Output:
+```text
+Running job 1 at 2020-01-02 01:11:00
+Running job 1 at 2020-01-02 01:11:00
+Running job 1 at 1970-01-01
+Running job 1 at 1970-01-01
+```
 
-job1, job2, job3, job4 = Job('1'), Job('2'), Job('3'), Job('4')
+The `Workflow.run` method ignores job parameters like `retry_count` and `retry_pause_sec`. It executes the workflow in a sequential (non-parallel) way.
```

Comment:
It executes the workflow ->  It executes a workflow

Reply:
done